dotnet/samples/TeamsChefBot/TeamsChefBotApplication.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.Bot.Builder;
+using Microsoft.TeamsAI;
+using Microsoft.TeamsAI.AI.Action;
+using Microsoft.TeamsAI.AI.Planner;
+using Microsoft.TeamsAI.State;
+using Microsoft.Bot.Schema;
+using System.Text.Json;
+
+namespace TeamsChefBot
+{
+    public class TeamsChefBotApplication : Application<TurnState, TurnStateManager>
+    {
+        public TeamsChefBotApplication(ApplicationOptions<TurnState, TurnStateManager> options) : base(options) 
+        {
+            AI.ImportActions(new TeamsChefBotActions());
+        }
+
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, TurnState turnState, CancellationToken cancellationToken)
+        {
+            if (turnContext.Activity.Type == ActivityTypes.Message)
+            {
+                if (turnContext.Activity.Text == "/history")
+                {
+                    string history = ConversationHistory.ToString(turnState, 2000, "\n\n");
+                    await turnContext.SendActivityAsync(history);
+                    return;
+                }
+            }
+            throw new NotImplementedException();
```

Comment:
This can be misleading. A better way to implement this would be to call `AI.ChainAsync("Chat")` if `text != "/history"`.

Reply:
Good idea. Thanks. Updated.